### PR TITLE
initializeMeshSubsets() across process vars and components.

### PR DESCRIPTION
**Pull request by endJunction**
One step towards multi-variable processes
MeshSubset initialization across all process variables.

### CODE REVIEW

File: ProcessLib/Process.h [hunk 1]
```diff
@@ -143,15 +144,21 @@ class Process
...
 
 		// Collect the mesh subsets in a vector.
-		_all_mesh_subsets.push_back(
-		    new MeshLib::MeshSubsets(_mesh_subset_all_nodes));
+		std::generate_n(
+		    std::back_inserter(_all_mesh_subsets),
+		    variable.getNumberOfComponents(),
+		    [&]()
+		    {
+			    return new MeshLib::MeshSubsets(_mesh_subset_all_nodes);
```
Comment (chleh) [inline ProcessLib/Process.h]: can we now also switch to unique_ptr?

**Comment by chleh** [comment]
:fast_forward: 
Could you please at some point (e.g., in a new issue) collect, what has to be done until multi-component process-variables will be completely supported (e.g., reading BCs, ICs, output (there was something still missing)).

File: ProcessLib/Process.h (hunk 1 above)
Comment (endJunction) [inline ProcessLib/Process.h]: This is a tricky one; Who has the ownership of the MeshSubsets, the Process or the DOF table?
It seems, like it should be the DOF table.

Comment (wenqing) [inline ProcessLib/Process.h]: Yes, should be DOF table.

Comment (chleh) [inline ProcessLib/Process.h]: But currently obviously `_all_mesh_subsets` owns them. So in the simplest case it is the question if you want to change the type of `_all_mesh_subsets` to `std::vector<std::unique_ptr<...>>`. Of course, then you also have to change some method signatures accordingly. Later on the meshsubsets may be owned by the DOF table. I don't care for now.
The problem I see is that most algorithms don't care if they work on a `std::vector<T*>` or on a `std::vector<std::unique_ptr<T>>` so maybe we should think about how to handle that without templating everything, but maybe I'm thinking too far.

**Comment by bilke** [comment]
:+1:

**Comment by endJunction** [comment]
Rebasing and merging then...

File: ProcessLib/Process.h (hunk 1 above)
Comment (endJunction) [inline ProcessLib/Process.h]: This is WIP @ https://github.com/endJunction/ogs/tree/UniquePtrMeshSubsets

**Pull request merged**